jdk/src/jdk.jdwp.agent/share/native/libjdwp/debugInit.c

```diff
@@ -779,6 +839,7 @@ debugInit_reset(JNIEnv *env)
 
     currentSessionID++;
     initComplete = JNI_FALSE;
+    VMInitComplete = JNI_TRUE; /* The VM Is already initialized */
```

Comment:
If this is done without entering the debug monitor, another thread may not see this variable update.

Reply:
If VMInitComplete is never achieved, couldn't a thread get stuck in debugInit_waitVMInitComplete() ?

Reply:
@pdbain-ibm Any update on these comments? i'm keen to get this clean before attempting to contribute to OpenJDK? thanks

Reply:
> If this is done without entering the debug monitor, another thread may not see this variable update

Could you elaborate? There are other shared variables which  are updated without entering monitors.

> If VMInitComplete is never achieved, couldn't a thread get stuck

VMInitComplete failure is an error condition, so we we should not proceed.

Reply:
Can debugInit_reset() be called while another thread is in debugInit_waitVMInitComplete()? If so debugInit_waitVMInitComplete() will be stuck in debugMonitorWait().

Reply:
debugInit_reset() is called only from within the debugLoop_run() loop. A thread calls debugInit_waitVMInitComplete() before entering said loop.

Reply:
What is the point of setting VMInitComplete to true in debugInit_reset() if nothing is going to use the variable from that point? Plus wouldn't it already have to be true anyway to be in the debugLoop_run() loop?

Reply:
My thoughts exactly.  Removed redundant assignment.